environment-dev.yml

```diff
@@ -0,0 +1,10 @@
+# conda env create -f environment-dev.yml
```

Comment:
That's a good point actually regarding `tests/requirements.txt`. 
We don't need to redefine them or making sure they are kept in sync. 
An entry can be added like so:


```yaml
dependencies:
- <conda-package-deps>
- pip:
  - -r tests/requirements.txt
```

Reply:
I think I will refer to the `tests/requirements.txt` file from my `environment-dev.yml`.  This leave the flexibility for someone preferring pip to conda.

I like conda because it can manage all dependencies, even system dependencies that are not python.  Pip can only exclusively manage python dependencies.  However pip is much more lightweight than conda.  So I do not want to force everyone to conda.  

As for the dependencies that seems to come from https://github.com/bird-house/birdhouse-deploy/pull/329, that's because I needed an env for that PR but decided to add this conda file here because this file cover more than just the testing dependencies so I feel more appropriate to add to this PR.  Anyways, if my conda file refer to the `tests/requirements.txt` file then this is not an issue anymore :)